platform/default/sqlite3.cpp

```diff
@@ -190,25 +187,46 @@ Database &Database::operator=(Database &&other) {
 
 Database::~Database() = default;
 
+
+DatabaseImpl::~DatabaseImpl() {
+    // Mark all transactions as invalid by removing the pointer to the database
```

Comment:
When does this situation arise? Do we not want it to be considered a logic error, and require the database to outlive any transactions?